src/common/components/CustomIcon/SchoolIcon/BaseIcon.tsx

```diff
@@ -0,0 +1,23 @@
+import { CustomIcon, type CustomIconProps } from "../CustomIcon";
+import { NTUIcon } from "@/common/components/CustomIcon/SchoolIcon/NTUIcon";
+import { NUSIcon } from "@/common/components/CustomIcon/SchoolIcon/NUSIcon";
+import { SMUIcon } from "@/common/components/CustomIcon/SchoolIcon/SMUIcon";
+
+export interface SchoolIconProps extends CustomIconProps {
+  school: "NTU" | "SMU" | "NUS";
+}
+const svgs = {
+  SMU: SMUIcon,
+  NUS: NUSIcon,
+  NTU: NTUIcon,
+};
+
+export const SchoolIcon = ({ school, ...props }: SchoolIconProps) => {
+  const schoolSVG = svgs[school] || SMUIcon;
+
+  return (
+    <CustomIcon viewBox="0 0 63 63" fill="none" {...props}>
+      {schoolSVG}
+    </CustomIcon>
+  );
+};

```

Comment:
Should we rename this file as `SchoolIcon`? Since the function exports SchoolIcon, I think it's clearer which file we should be finding!

Reply:
amended and thanks!